electron-app/src/views/settings/Settings.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <v-container class="settings">
+    <v-row>
+      <v-col>
+        <h1>Settings</h1>
+        <tab-navigation :tab-contents="settingsTabs"></tab-navigation>
+      </v-col>
+    </v-row>
+    <v-slide-x-transition>
+      <router-view />
+    </v-slide-x-transition>
+  </v-container>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator';
+import TabNavigation from '@/components/helper/TabNavigation.vue';
+
+export interface SettingsMessages {
+  [index: string]: {
+    success: string;
+    error: string;
+  };
+}
+
+@Component({
+  components: { TabNavigation }
+})
+export default class Settings extends Vue {
+  settingsTabs = [
+    {
+      name: 'General',
+      routerTo: '/settings/general'
+    },
+    {
+      name: 'Accounting',
+      routerTo: '/settings/accounting'
+    },
+    {
+      name: 'User & Security',
+      routerTo: '/settings/usersecurity'
+    }
+  ];
+
+  settingsMessages: SettingsMessages = {};
+
+  validateSettingChange(
+    targetSetting: string,
+    targetState: string,
+    message: string = '',
+    timeOut: number = 10000
+  ) {
+    if (targetState === 'success' || targetState === 'error') {
+      setTimeout(() => {
+        this.settingsMessages[targetSetting][targetState] = `Setting${
+          targetState === 'error' ? ' not ' : ' '
+        }saved${message !== '' ? ': ' + message : ''}`;
```

Comment:
:cry:

Reply:
Do you dislike the approach of abstracting the validation of the event or just the ternary-chaos?  If it's the latter should I just do all the resolution of the messages in ifs above and then just print a cleaner string?

Reply:
yes maybe it would make sense to extract anything needed and not have ternary operators in the pattern